src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -132,23 +132,21 @@ trait IndexedTable {
   def optimize(revisionID: RevisionID, fraction: Double, options: Map[String, String]): Unit
 
   /**
-   * Optimizes the given table for a given revision
-   * @param revisionID
-   *   the identifier of revision to optimize
+   * Optimizes the table by optimizing the data stored in the specified index files.
+   *
+   * @param indexFiles
+   *   the index files to optimize
    * @param options
    *   Optimization options where user metadata and pre-commit hooks are specified.
    */
-  def optimize(revisionID: RevisionID, options: Map[String, String]): Unit
+  def optimizeIndexFiles(indexFiles: Seq[String], options: Map[String, String]): Unit
 
   /**
-   * Optimizes the table by optimizing the data stored in the specified index files.
-   *
-   * @param files
-   *   the index files to optimize
+   * Optimizes the table by optimizing the data stored in the specified unindexed files.
+   * @param unindexedFiles
    * @param options
-   *   Optimization options where user metadata and pre-commit hooks are specified.
    */
-  def optimize(files: Seq[String], options: Map[String, String]): Unit
+  def optimizeUnindexedFiles(unindexedFiles: Seq[String], options: Map[String, String]): Unit
```

Comment:
Another concern is the terminology; UNindexed files sound more like files that were indexed at one point but unindexed. What about `nonIndexedFiles?`

Reply:
sure

Reply:
In all cases I think they are Unindexed:
- For appends and legacy: the data is never indexed with Qbeast
- For updates and deletes: the content of the new file might have been indexed in the past, but now it's a different file, so it's never indexed with Qbeast

I know that the reasoning might conflict logically in point 2, but... is it convincing?

Reply:
after discussion, we decided that, since updates and deletes are outside of clarification for Unindexed, we should keep the implementation with the current naming. 

Also, Unindexed means Non-Indexed, so there should be no problem with terminology.